text/0431-sagemaker-l2-endpoint.md

```diff
@@ -0,0 +1,1150 @@
+     * registry/repository@digest.
+     *
+     * For example, 012345678910.dkr.ecr.<region-name>.amazonaws.com/<repository-name>:latest or
+     * 012345678910.dkr.ecr.<region-name>.amazonaws.com/<repository-name>@sha256:94afd1f2e64d908bc90dbca0035a5b567EXAMPLE.
+     */
+    readonly imageName: string;
+  }
+  ```
+
+- `ContainerImage` -- abstract class defining `bind` contract for images alongside static factory
+  methods to enable different sources (e.g., image in ECR repository, local Dockerfile)
+
+  ```ts
+  export abstract class ContainerImage {
+    /**
+     * Reference an image in an ECR repository
+     */
+    public static fromEcrRepository(repository: ecr.IRepository, tag: string = 'latest'): ContainerImage { ... }
+
+    /**
+     * Reference an image that's constructed directly from sources on disk
+     *
+     * @param scope The scope within which to create the image asset
+     * @param id The id to assign to the image asset
+     * @param props The properties of a Docker image asset
+     */
+    public static fromAsset(scope: Construct, id: string, props: assets.DockerImageAssetProps): ContainerImage { ... }
+
+    /**
+     * Called when the image is used by a Model
+     */
+    public abstract bind(scope: Construct, model: Model): ContainerImageConfig;
+  }
+  ```
+
+###### Model Data
+
+Analogous to the above pairing of `ContainerImageConfig` and `ContainerImage`, the following
+interface and abstract class provide mechanisms for customers to specify the source of their model
+artifacts, either in an S3 bucket or a local file asset.
+
+- `ModelDataConfig` -- the configuration needed to reference model artifacts
+
+  ```ts
+  export interface ModelDataConfig {
+    /**
+     * The S3 path where the model artifacts, which result from model training, are stored. This path
+     * must point to a single gzip compressed tar archive (.tar.gz suffix).
+     */
+    readonly uri: string;
+  }
+  ```
+
+- `ModelData` -- model data represents the source of model artifacts, which will ultimately be
+  loaded from an S3 location
+
+  ```ts
+  export abstract class ModelData {
+    /**
+     * Constructs model data which is already available within S3.
+     * @param bucket The S3 bucket within which the model artifacts are stored
+     * @param objectKey The S3 object key at which the model artifacts are stored
+     */
+    public static fromBucket(bucket: s3.IBucket, objectKey: string): ModelData { ... }
+
+    /**
+     * Constructs model data that will be uploaded to S3 as part of the CDK app deployment.
+     * @param scope The scope within which to create a new asset
+     * @param id The id to associate with the new asset
+     * @param path The local path to a model artifact file as a gzipped tar file
+     */
+    public static fromAsset(scope: Construct, id: string, path: string): ModelData { ... }
```

Comment:
Noted above but copying here as well:

> In rev2, I've pivoted to having the new proposed APIs take an `Asset` from '@aws-cdk/aws-s3-assets' (for `ModelData`) and `DockerImageAsset` from '@aws-cdk/aws-ecr-assets' (for `ContainerImage`) to avoid adding any assets into the hierarchy myself.